packages/suite-desktop/src/libs/processes/BaseProcess.ts

```diff
@@ -133,13 +133,31 @@ export abstract class BaseProcess {
             `- Params: ${params}`,
             `- CWD: ${processDir}`,
         ]);
-        this.process = spawn(processPath, params, {
-            cwd: processDir,
-            env: processEnv,
-            stdio: ['ignore', 'ignore', 'ignore'],
+
+        return new Promise<void>((resolve, reject) => {
+            const newProcess = spawn(processPath, params, {
+                cwd: processDir,
+                env: processEnv,
+                stdio: ['ignore', 'ignore', 'ignore'],
+            });
+            newProcess.on('error', err => this.onError(err));
+            newProcess.on('exit', code => this.onExit(code));
+
+            let resolveTimeout: ReturnType<typeof setTimeout> | undefined;
+            const spawnErrorHandler = (message: any) => {
+                clearTimeout(resolveTimeout);
+                reject(new Error(`Process ${this.processName} not started. ${message}`));
+            };
+
+            newProcess.once('error', spawnErrorHandler);
+            newProcess.once('exit', spawnErrorHandler);
+
+            resolveTimeout = setTimeout(() => {
+                newProcess.removeListener('exit', spawnErrorHandler);
+                newProcess.removeListener('error', spawnErrorHandler);
+                resolve();
+            }, 100);
```

Comment:
isn't there a better way than this timeout? 
- maybe this https://nodejs.org/api/child_process.html#event-spawn
- or maybe revisit `stdio: ['ignore', 'ignore', 'ignore']` and find a way how to consume stdout without killing app memory?

Reply:
I updated it to use `this.status()` and it works you can check it https://github.com/trezor/trezor-suite/pull/6528/commits/f0522182bd69ad3a84098c3f43f7762fc803e2f3

It makes it slower but it seams the right way. WDYT?

I rebased the whole thing.